datafusion/core/src/datasource/listing/url.rs

```diff
@@ -317,8 +315,27 @@ mod tests {
         let url = ListingTableUrl::parse("file:///foo/bar?").unwrap();
         assert_eq!(url.prefix.as_ref(), "foo/bar");
 
-        let url = ListingTableUrl::parse("file:///foo/😺").unwrap();
-        assert_eq!(url.prefix.as_ref(), "foo/%F0%9F%98%BA");
+        let err = ListingTableUrl::parse("file:///foo/😺").unwrap_err();
```

Comment:
It seems very reasonable to treat urls like `file://foo`, as urls (rather than paths with a `file://` prefix)